js/reference.js

```diff
@@ -6595,3 +6618,126 @@
             }
         });
     }
+
+    /**
+     *
+     * @param {ERMrest.Reference} reference reference for the association table
+     * @param {Object} prefillObject computed prefill object after extracting the query param and fetching the data from cookie storage
+     * @param {Object} prefillAnnotation
+     */
+    function Prefill (reference, prefillObject, prefillAnnotation) {
+        verify(prefillObject, "'prefillObject' must be defined");
+
+        var self = this;
+        this._reference = reference;
+        this._prefillObject = prefillObject;
+        this._prefillAnnotation = prefillAnnotation;
+
+        // ignore the fks that are simple and their constituent column is system col
+        var nonSystemColumnFks = this._reference.table.foreignKeys.all().filter(function(fk) {
+            return !(fk.simple && module._systemColumns.indexOf(fk.colset.columns[0]) !== -1);
+        });
+
+        // set of foreignkey columns (they might be overlapping so we're not using array)
+        var fkCols = {};
+        nonSystemColumnFks.forEach(function(fk) {
+            fk.colset.columns.forEach(function(col) {
+                fkCols[col] = true;
+            });
+        });
+
+        // the following are ERMrest.ForeignKeyPseudoColumn
+        this.mainColumn;
+        this.leafColumn;
+
+         // NOTE: I think this can be done differently. Leaving this alone for now for the initial implementation
+        reference.columns.forEach(function(column) {
+            // column should be a foreignkey pseudo column
+            if (!column.isForeignKey) return;
+
+            nonSystemColumnFks.forEach(function(fk) {
+                // column and foreign key `.name` property is a hash value
+                if (column.name === fk.name) {
+                    if (prefillObject.fkColumnNames.indexOf(column.name) !== -1) {
+                        self.mainColumn = column;
+                    } else {
+                        // TODO: what if there are multiple "other" fk columns?
```

Comment:
I briefly discussed this with Hongsuda yesterday in standup. She stated at that point it's no longer an association since we don't have an annotation defined yet to "assist" in figuring this out.